tests/ci/azureml_tests/post_pytest.py

```diff
@@ -0,0 +1,90 @@
+# Copyright (c) Recommenders contributors.
```

Comment:
@SimonYansenZhao I think something that is missing is to be able to see the tests that have been executed in the logs: https://github.com/recommenders-team/recommenders/actions/runs/10157874198/job/28088807068

In previous runs, https://github.com/recommenders-team/recommenders/actions/runs/10103296252/job/27940415983
we can see the tests like this:
```
INFO:submit_groupwise_azureml_pytest.py:Executing tests now...
============================= test session starts ==============================
platform linux -- Python 3.8.18, pytest-8.2.2, pluggy-1.5.0
rootdir: /mnt/azureml/cr/j/2953317441644d75953b7e5c220b3657/exe/wd
configfile: pyproject.toml
plugins: cov-5.0.0, mock-3.14.0, typeguard-4.3.0, hypothesis-6.105.1, anyio-4.4.0
collected 3 items

tests/unit/examples/test_notebooks_gpu.py .                              [ 33%]
tests/smoke/recommenders/recommender/test_newsrec_model.py 

============================== slowest durations ===============================
199.59s call     tests/smoke/recommenders/recommender/test_newsrec_model.py::test_model_nrms
106.89s call     tests/smoke/recommenders/recommender/test_newsrec_model.py::test_model_npa
1.62s call     tests/unit/examples/test_notebooks_gpu.py::test_gpu_vm
```

Reply:
Yes, I've noticed that.  I am trying to figure it out.  But currently, the pytest logs can be found in the artifacts of the Action runs.  See [the example run](https://github.com/recommenders-team/recommenders/actions/runs/10157874198) below when you scroll down to the page bottom of the run:
<img width="1707" alt="Screenshot 2024-07-30 at 20 28 35" src="https://github.com/user-attachments/assets/f34b20f1-4063-41ca-95be-58c3573dd381">

Reply:
I think what we did in the past was download it and show it in the logs.

Reply:
Makes sense, I think it was available after the tests finished, but you could not see the progress.

Reply:
After research, I find the pytest logs were streamed by AML SDK.

What we did to pytest when using AML SDK v1 is shown in Figure 1, and what the logs look like is shown in Figure 2.  Now the code in AML SDK v2 seems to print all logs (See Figure 3), but the actual results is shown in Figure 4 where no pytest logs was output.

So I guess this may be a bug in AML SDK v2!
I'll add code to print the pytest logs in a separate step in the workflow, and remove it when they fix the bug in the future.

Figure 1

<img width="674" alt="Screenshot 2024-07-30 at 22 22 00" src="https://github.com/user-attachments/assets/269499c8-806a-46ff-ae7a-9ac3fae37f5f">

Figure 2

<img width="1035" alt="Screenshot 2024-07-30 at 22 22 28" src="https://github.com/user-attachments/assets/cec93616-7135-4b17-92b0-597a6cf622df">

Figure 3

![Screenshot 2024-07-30 at 22 24 40](https://github.com/user-attachments/assets/57996c4b-0b6d-4bd6-a5c9-0efc833517bd)

Figure 4

<img width="989" alt="Screenshot 2024-07-30 at 22 29 30" src="https://github.com/user-attachments/assets/79705940-e011-4042-935a-70f0d6da6397">

Reply:
BTW, I also find the docker used in AML service is of some version back in 2021 😂 because it didn't recognize [heredocs](https://docs.docker.com/reference/dockerfile/#here-documents) supported in [Dockerfile version 1.4.0 (released in 2022.3)](https://docs.docker.com/build/dockerfile/release-notes/#140) when I tried to replace all `&& \` multi-line command with heredocs.

Reply:
> I'll add code to print the pytest logs in a separate step in the workflow, and remove it when they fix the bug in the future.


If you could do it in a separate step, it would be nice because in the Execute tests we would be able to see if there is an error in the build process, and have another step with the results of the tests and we only see if there was an error in the tests. 

![Uploading image.png…]()

Right now it is very messy because we have both the build and the test in the same section

Reply:
Done

Reply:
Now the pytest logs are folded.  See https://github.com/recommenders-team/recommenders/actions/runs/10173799542/job/28139368861?pr=2134#step:3:1135

<img width="981" alt="Screenshot 2024-07-31 at 13 10 49" src="https://github.com/user-attachments/assets/f9283b03-d75e-4ab5-81fd-5d9b028468d9">